### DISCUSSION TITLE
security tools: GnuPG, SSL tools

### DISCUSSION THREAD

**Discussion by edwinjhlee** [jart/cosmopolitan#1207] [Ideas]
CosmoCC is well-suited to compile and provide security tools. These tools are small, essential, and algorithm-based, making them ideal for integration.

Are there any plans to include such tools in Cosmo?

**Comment by jart** [comment]
Does GCC integrate GnuPG?

**Reply by edwinjhlee** [reply]
> Does GCC integrate GnuPG?

Sorry. I am not good at English. I might use the word 'integrate' by mistake.

I am just suggesting cosmopolitan official can consider providing security tools in 
https://cosmo.zip/pub/cosmos/bin/  and  https://github.com/jart/cosmopolitan/tree/master/tool

**Comment by mrdomino** [comment]
I have been sitting here thinking about doing something like that for a bit now.

E.g. a command-line argon2 swiss-army knife would allow me to do some stuff from shell that you probably shouldn't do from shell.

I think maybe the question you wanted to ask is: "what if someone made something like the openssl command-line tool but that wasn't bad, and then you had a cosmopolitan-compiled distribution of it that you could use everywhere?"

(Though, come to think of it - @ahgamut superconfigure already builds the openssl library, right? How hard would it be to get the CLI tool out of it as well?)

Anyway - to do something nearly as comprehensive as openssl would be a lot of work. I think probably targeting smaller, discrete use cases (like e.g. [passgen](https://github.com/xfbs/passgen)) is going to be a higher payoff most of the time.
